core/store/src/lib.rs

```diff
@@ -86,8 +87,25 @@ pub struct Store {
 
 impl NodeStorage {
     /// Initialises a new opener with given home directory and store config.
-    pub fn opener<'a>(home_dir: &std::path::Path, config: &'a StoreConfig) -> StoreOpener<'a> {
-        StoreOpener::new(home_dir, config)
+    ///
+    /// If the database is not of the exact version that we’re expecting, the
+    /// opening will fail.  No migration will be attempted.  If you want to
+    /// perform migrations when opening a database, use
+    /// [`Self::opener_with_migrator`].
+    pub fn opener<'a>(home_dir: &std::path::Path, config: &'a StoreConfig) -> StoreOpener<'a, ()> {
+        StoreOpener::new(home_dir, config, ())
+    }
+
+    /// Initialises a new opener with given database migrator.
+    ///
+    /// When database is open and it has incorrect database version, migration
+    /// will be attempted with the help of the specified `migrator`.
+    pub fn opener_with_migrator<'a, M: StoreMigrator>(
```

Comment:
I'd add `.with_migrator(self, M) -> StoreOpener<'a, M>` method to `StoreOpener<()>` instead.